qiskit_experiments/framework/config.py

```diff
@@ -0,0 +1,144 @@
+    :class:~qiskit_experiments.framework.BaseExperiment` and
+    can be used to reconstruct the experiment using either the
+    :meth:`experiment` property if the experiment class type is
+    currently stored, or the
+    :meth:~qiskit_experiments.framework.BaseExperiment.from_config`
+    class method of the appropriate experiment.
+    """
+
+    cls: type = None
+    args: Tuple[Any] = dataclasses.field(default_factory=tuple)
+    kwargs: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    experiment_options: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    transpile_options: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    run_options: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    version: str = __version__
+
+    @property
+    def experiment(self):
+        """Return the experiment constructed from this config.
+
+        Returns:
+            BaseExperiment: The experiment reconstructed from the config.
+
+        Raises:
+            QiskitError: if the experiment class is not stored,
+                         was not successful deserialized, or reconstruction
+                         of the experiment fails.
+        """
+        cls = self.cls
+        if cls is None:
+            raise QiskitError("No experiment class in experiment config")
+        if isinstance(cls, dict):
+            raise QiskitError(
+                "Unable to load experiment class. Try manually loading "
+                "experiment using `Experiment.from_config(config)` instead."
+            )
+        try:
+            return cls.from_config(self)
+        except Exception as ex:
+            msg = "Unable to construct experiments from config."
+            if cls.version != __version__:
+                msg += (
+                    f" Note that config version ({cls.version}) differs from the current"
+                    f" qiskit-experiments version ({__version__}). You could try"
+                    " installing a compatible qiskit-experiments version."
+                )
+            raise QiskitError("{}\nError Message:\n{}".format(msg, str(ex))) from ex
+
+
+@dataclasses.dataclass(frozen=True)
+class AnalysisConfig:
+    """Store configuration settings for Analysis
+
+    This stores the current configuration of a
+    :class:~qiskit_experiments.framework.BaseAnalysis` and
+    can be used to reconstruct the analysis class using either the
+    :meth:`analysis` property if the analysis class type is
+    currently stored, or the
+    :meth:~qiskit_experiments.framework.BaseAnalysis.from_config`
+    class method of the appropriate experiment.
+    """
+
+    cls: type = None
+    args: Tuple[Any] = dataclasses.field(default_factory=tuple)
+    kwargs: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    options: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    version: str = __version__
+
+    @property
+    def analysis(self):
```

Comment:
Why this property is necessary?

Reply:
If config is complete this lets you convert it to analysis class instance without having to look up and import that analysis class.